song-server/src/main/java/org/icgc/dcc/song/server/service/UploadService.java

```diff
@@ -63,7 +65,12 @@
   private final UploadRepository uploadRepository;
 
   public Upload read(@NonNull String uploadId) {
-    return uploadRepository.get(uploadId);
+    val upload = uploadRepository.get(uploadId);
```

Comment:
We should try to put client changes and server changes into separate PRs, if only for tracking purposes.